### DISCUSSION TITLE
Pesan yang dikirim masuk kembali sebagai incoming_message.

### DISCUSSION THREAD

**Discussion by fahram** [kirimwa-id/forum#60] [Tanya Jawab] [answered]
Permisi lagi Bang Rio. 
Mungkin ini bug juga. 
Setiap pesan yang saya kirim menggunakan end point : `POST /messages`
akan masuk ke Webhook Incoming Message, jadi setiap saya kirim pesan maka akan ada 3 respon ke webhook (**1 send_message_response, 2 incoming_message** ) 

### sebagai contoh : 

> Message Response 

<pre><code>  'id' => 'kwid-046b61bdcf52476ab17a29c9fe9',
  'status' => 'success',
  'message' => 'Message has been sent.',
  'webhook_type' => 'send_message_response',
  'payload' =>
  array (
    'message' => 'Test post message',
    'phone_number' => '62816689***,
    'device_id' => '6281748564***',
    'message_type' => 'text',
    'caption' => NULL,
    'is_group_message' => false,
  ),
  'created_at' => '2021-10-29T05:51:12.952Z',
  'server_time' => '2021-10-29T05:51:13.035Z',</pre></code>

> Incoming 1: 

<pre><code> 'id' => 'kwid-046b61bdcf52476ab17a29c9fe9',
  'webhook_type' => 'incoming_message',
  'payload' =>
  array (
    'id' => 'kwid-046b61bdcf52476ab17a29c9fe9',
    'device_id' => '628174856***',
    'sender' => '628174856***',
    'from_me' => true,
    'text' => 'Test post message',
    'caption' => NULL,
    'message_type' => 'text',
    'is_group_message' => false,
    'group_id' => NULL,
    'timestamp' => 1635486673,
    'contact' =>
    array (
      'name' => NULL,
      'vcard' => NULL,
    ),
    'location' =>
    array (
      'latitude' => 0,
      'longitude' => 0,
      'name' => NULL,
      'address' => NULL,
    ),
    'meta' =>
    array (
      'file_name' => NULL,
      'seconds' => 0,
      'size' => 17,
      'mime_message_type' => NULL,
      'dimension' =>
      array (
        'height' => 0,
        'width' => 0,
      ),
      'mime_type' => 'plain/text',
    ),
  ),
  'created_at' => '2021-10-29T05:51:13.000Z',
  'server_time' => '2021-10-29T05:51:13.841Z',</pre></code>

>  Incoming 1: 
 

<pre><code>  'id' => 'kwid-046b61bdcf52476ab17a29c9fe9',
  'webhook_type' => 'incoming_message',
  'payload' =>
  array (
    'id' => 'kwid-046b61bdcf52476ab17a29c9fe9',
    'device_id' => '628174856***',
    'sender' => '628174856***',
    'from_me' => true,
    'text' => 'Test post message',
    'caption' => NULL,
    'message_type' => 'text',
    'is_group_message' => false,
    'group_id' => NULL,
    'timestamp' => 1635486673,
    'contact' =>
    array (
      'name' => NULL,
      'vcard' => NULL,
    ),
    'location' =>
    array (
      'latitude' => 0,
      'longitude' => 0,
      'name' => NULL,
      'address' => NULL,
    ),
    'meta' =>
    array (
      'file_name' => NULL,
      'seconds' => 0,
      'size' => 17,
      'mime_message_type' => NULL,
      'dimension' =>
      array (
        'height' => 0,
        'width' => 0,
      ),
      'mime_type' => 'plain/text',
    ),
  ),
  'created_at' => '2021-10-29T05:51:13.000Z',
  'server_time' => '2021-10-29T05:51:14.662Z',</pre></code>

**Comment by rioastamal** [comment, marked as answer]
Betul, ini sudah teridentifikasi beberapa waktu lalu dan akan difix pada rilis berikutnya 1.0-RC9. Terima kasih.
